packages/multi-tenant/src/discoverTenant.ts

```diff
@@ -0,0 +1,38 @@
+import getMatch from "./lib/getMatch";
+import getMultiTenantConfig from "./lib/multiTenantConfig";
+import TenantService from "./model/tenants/service";
+
+import type { Tenant } from "./types";
+import type { FastifyReply, FastifyRequest } from "fastify";
+
+const discoverTenant = async (request: FastifyRequest, reply: FastifyReply) => {
```

Comment:
Preferably, use primitive as arguments in order to make unit testing easier. As per below, Reply should NOT be an argument anyway.
Instead of Request, use primitive arguments (eg headers, hostname).